api/v1alpha1/bmcjob_types.go

```diff
@@ -0,0 +1,126 @@
+/*
+Copyright 2022 Tinkerbell.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// BMCJobConditionType represents the condition of the BMC Job.
+type BMCJobConditionType string
+
+const (
+	// JobCompleted represents successful completion of the BMC Job tasks.
+	JobCompleted BMCJobConditionType = "Completed"
+	// JobFailed represents failure in BMC job execution.
+	JobFailed BMCJobConditionType = "Failed"
+	// JobTaskCompleted represents successful completion of a BMC Job task.
+	JobTaskCompleted BMCJobConditionType = "TaskCompleted"
+)
+
+// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
+
+// BMCJobSpec defines the desired state of BMCJob
+type BMCJobSpec struct {
+	// Tasks represents a list of baseboard management actions to be executed.
+	// The tasks are completed in order.
```

Comment:
We need commentary on whether we wait for the outcome of a task before proceeding with the next and whether we proceed to the next if the first fails.

Reply:
Sure! I can add that. Right now my plan was

1. Wait for a task to complete before moving to next
2. If single task fails, we set the job as `Failed`
3. We only set `Completed` when all the tasks have completed successfully.

This does pose an argument where there can be cases in which we want to power off/reset even if the previous operation was not successful. Even maybe have retries? But I did not scope those for now.